packages/api/src/external/carequality/process-subsequent-outbound-patient-discovery-resps.ts

```diff
@@ -44,7 +44,6 @@ export async function processPostRespOutboundPatientDiscoveryResps({
   } catch (error) {
     const msg = `Error on Post Resp Outbound PD Responses`;
     log(`${msg} - ${errorToString(error)}`);
-    await processPatientDiscoveryProgress({ patient: { id: patientId, cxId }, status: "failed" });
```

Comment:
Removed as we shouldn't set the entire PD process to failed if only one post from the GW fails.

Reply:
these status things are finnicky. Are we sure this wont introduce unexpected issues? Is this a change being made in response to an observed bug or odd behavior or just based on reasoning?

Reply:
Based on reasoning. If a single POST response failed in this code (and the PD status gets flipped to `failed`), the next successful POST response would trigger another lambda unnecessarily as it would find the overall state of the flow as not `processing` (even though it is). Generally, this status represents the success / failure of the entire flow, not handling of an individual POST response via this function, so flagging a failure here isn't representative of what's happening. If this was happening in the background, I don't think we'd know as DQ is scheduled only once and the lambdas would all perform the same 

I put it back in for now and flagged like the rest.

Reply:
Topic to another issue, but I think we could have a state machine to update the status, with inputs being new errors, new successes, etc.

I suggest we look at the commit that introduced this update. If it was a bug fix, fine to leave it. If it was there since the beginning, I'd consider removing it if that makes sense - which I think it does based on Thomas' comment.

Reply:
@leite08 I feel like the PR has enough changes as is that we should flag it as part of a larger "state handling" issue for PD. If we move forward with the scheduling edits I've made in this PR, then this `processPostRespOutboundPatientDiscoveryResps` is really the only non-admin trigger place for the PD flow. `discover` / `create` / `update` are being called directly in `patient-updater` in both CQ / CW and `recreate-patients-at-hies` in CW -- but my sense is that we control when those get run.

Reply:
Larger state handling issue https://github.com/metriport/metriport-internal/issues/1832 (put a TODO in the code for each place right now we enter the PD flow outside of the scheduling framework).